### ISSUE TITLE
Rustdoc does not show documentation on empty impl blocks

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#90866]
I tried this code:

```rust
struct Ty {}
/// Documentation 1
impl Ty {}
```

I expected to see this happen: Either the comment should show up in the generated docs, or rustdoc/rustc should issue a warning.

Instead, this happened: The documentation is silently ignored.

I think we should show the documentation, since people like to use impl blocks for grouping and it's possible that all the items in the impl have a `#[cfg()]` macro that's false.

Thanks @WaffleLapkin for the bug report and @GuillaumeGomez for giving feedback on the proper fix.

### Meta

`rustdoc --version --verbose`: rustdoc 1.58.0-nightly (e90c5fbbc 2021-11-12)

**Comment by GuillaumeGomez** [comment]
I'll write the fix in the next days.

**Comment by kpreid** [comment]
A suggestion: Since this is such an unusual case, add a special note like

> _This `impl TypeName {}` contains no items._

after the displayed documentation. This will help people parse the structure of the documentation page, understand what it _means_ that the thing is said to be an implementation, and understand that it's not purely a rustdoc bug/infelicity that it's showing an empty section.

**Comment by GuillaumeGomez** [comment]
Good idea!
